sdk/src/proto/network.rs

```diff
@@ -1,125 +1,234 @@
 // This file is @generated by prost-build.
+/// The request for a nonce. Should be called before signing a message to get the correct nonce for
```

Comment:
nit: Should only -> Returns an error if

Reply:
Arguably it's better to follow https://www.ietf.org/rfc/rfc2119.txt standard but Im down to change it to error conditions instead.

Reply:
so yeah something like this `Claim a requested proof, commiting the proving and later fulfilling it. MUST be called proof is in a PROOF_REQUESTED state.` on the RPC spec, lmk if you have any issues with that